websites/D/Doordash/presence.ts

```diff
@@ -0,0 +1,101 @@
+const presence = new Presence({
+		clientId: "1001831947611426877",
+	}),
+	started = Date.now();
+
+presence.on("UpdateData", async () => {
+	const path: string = document.location.pathname,
+		presenceData: PresenceData = {
+			largeImageKey: "logo",
+			startTimestamp: started,
+		};
+
+	// If they're using includes statements, it means the page is not a static page (like home, orders, etc.)
+
+	// Store page
+	if (path.includes("/store/") && !path.includes("/search/")) {
+		const name = document.querySelector<HTMLSpanElement>(
+				"h1.styles__TextElement-sc-1jrzcv7-0"
+			).textContent,
+			image = document.querySelector<HTMLImageElement>(
+				".menu-header-styles__LogoInner-sc-1kgi6s8-6 > picture:nth-child(1) > img:nth-child(5)"
+			),
+			hideFood = await presence.getSetting<boolean>("hideFood");
+
+		if (
+			document.querySelector<HTMLImageElement>(
+				".styles__ModalContainer-sc-1r4qbfh-0"
+			) === null
+		) {
+			presenceData.largeImageKey = image.src;
+			presenceData.smallImageKey = "logo";
+			presenceData.details = `Viewing the menu of ${name}`;
+		} else {
+			if (!hideFood) {
+				const menuItemImage = document.querySelector<HTMLImageElement>(
+					".BaseItemModalStyles__PictureFrame-sc-mpq8va-1 > picture:nth-child(1) > img:nth-child(5)"
+				);
+				presenceData.state = `Looking at ${
+					document.querySelector<HTMLSpanElement>(".bbOfWk").textContent
```

Comment:
Tbh no reason to use this unless you're gonna acces things that don't exist on normal elements.
```suggestion
					document.querySelector(".bbOfWk").textContent
```

Reply:
Not exactly, you should ideally always specify the type for furture maintainability